### PR TITLE
feat: create tray launcher

### PR DESCRIPTION
**Pull request by alexandercbooth**
For users that consume mountains of notebooks, this might be helpful 😄 

Thoughts on menu items/structure/wording? Below are some screen shots. ⬇️ 

image

At the moment, I don't think the logo is a high enough resolution 😬

image

### CODE REVIEW

**Comment by rgbkrk** [comment]
Super cool! I think I can trim down that logo so it's only the icon and give you a higher resolution version. This gives a good idea of it though!

File: applications/desktop/src/main/index.js [hunk 1]
```diff
@@ -244,14 +251,19 @@ openFile$
...
 fullAppReady$.subscribe(() => {
   kernelSpecsPromise
     .then(kernelSpecs => {
       if (Object.keys(kernelSpecs).length !== 0) {
         store.dispatch(setKernelSpecs(kernelSpecs));
         const menu = loadFullMenu();
         Menu.setApplicationMenu(menu);
+        const trayImage = join(__dirname, "..", "static", "nteractlogo.png");
+        tray = new Tray(trayImage);
+        tray.setToolTip("nteract Quick Launch");
```
Comment (rgbkrk) [inline applications/desktop/src/main/index.js]: I was wondering what other apps and native mac utilities report in their tooltip and only my VPN shows one. Based on that, I'm not sure if we should.

My thinking here is totally counter to what I usually say with web tools and toolbars. 🤷‍♂️

File: applications/desktop/src/main/menu.js [hunk 2]
```diff
@@ -653,3 +653,128 @@ export function loadFullMenu(store: * = global.store) {
...
+        }
+      ]
+    });
+  }
+
+  template.push(fileWithNew);
+
+  const menu = Menu.buildFromTemplate(template);
+  return menu;
+}
```
Comment (rgbkrk) [inline applications/desktop/src/main/menu.js:780]: 👏

File: applications/desktop/src/main/index.js (hunk 1 above)
Comment (alexandercbooth) [reply applications/desktop/src/main/index.js]: That's a good point, none of the others on my machine show anything either. Will remove 👍

File: applications/desktop/src/main/menu.js [hunk 3]
```diff
@@ -653,3 +653,128 @@ export function loadFullMenu(store: * = global.store) {
   const menu = Menu.buildFromTemplate(template);
   return menu;
 }
+
+export function loadTrayMenu(store: * = global.store) {
+  // NOTE for those looking for selectors -- this state is not the same as the
+  //      "core" state -- it's a main process side model in the electron app
```
Comment (BenRussert) [inline applications/desktop/src/main/menu.js:659]: This is a good tip, maybe we should remember it for the [dev docs](https://github.com/nteract/global-sprint/blob/master/projects/docs.md#devdocsnteractio)

File: applications/desktop/src/main/index.js [hunk 4]
```diff
@@ -244,14 +251,18 @@ openFile$
...
-
+let tray = null;
 fullAppReady$.subscribe(() => {
   kernelSpecsPromise
     .then(kernelSpecs => {
       if (Object.keys(kernelSpecs).length !== 0) {
         store.dispatch(setKernelSpecs(kernelSpecs));
         const menu = loadFullMenu();
         Menu.setApplicationMenu(menu);
+        const trayImage = join(__dirname, "..", "static", "nteractlogo.png");
```
Comment (rgbkrk) [inline applications/desktop/src/main/index.js]: It looks like we can create multiple resolutions of this same image: https://electronjs.org/docs/api/native-image#high-resolution-image

**Comment by alexandercbooth** [comment]
## Update
- fixed logo and much sharper 👍 😎 
- included white logo for dark menu bar and windows

image


image

Thanks for the feedback and looking this over, @rgbkrk and @BenRussert!

**Comment by rgbkrk** [comment]
Slick.

**Comment by rgbkrk** [comment]
Now that I'm using this, can we make the launcher just be "New" and "Open" instead of "Launch New Notebook" then "New", "Open", etc.?

**Comment by alexandercbooth** [comment]
Certainly!

**Pull request merged**